src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp

```diff
@@ -679,19 +683,34 @@ bool PDPServer2::process_data_queue()
     return discovery_db_.process_data_queue();
 }
 
+void PDPServer2::awakeServerThread(
+        double interval_ms)
+{
+    mp_sync->update_interval_millisec(interval_ms);
+    mp_sync->restart_timer();
+}
+
 bool PDPServer2::server_update_routine()
 {
-    logInfo(RTPS_PDP_SERVER, "");
-    logInfo(RTPS_PDP_SERVER, "-------------------- Server routine start --------------------");
-    bool result = process_writers_acknowledgements(); // server + ddb(functor_with_ddb)
-    process_data_queue();                             // all ddb
-    result |= process_disposals();                    // server + ddb(changes_to_dispose(), clear_changes_to_disposes())
-    result |= process_dirty_topics();                 // all ddb
-    result |= process_to_send_lists();                // server + ddb(get_to_send, remove_to_send_this)
-    result |= process_changes_release();              // server + ddb(changes_to_release(), clear_changes_to_release())
-    result |= pending_ack();                          // all server
-    logInfo(RTPS_PDP_SERVER, "-------------------- Server routine end --------------------");
-    logInfo(RTPS_PDP_SERVER, "");
+    bool result = true;
+    // If the data queue is not empty re-start the routine
+
+    do
+    {
+        logInfo(RTPS_PDP_SERVER, "");
+        logInfo(RTPS_PDP_SERVER, "-------------------- Server routine start --------------------");
+        process_writers_acknowledgements();               // server + ddb(functor_with_ddb)
+        process_data_queue();                             // all ddb
+        process_disposals();                              // server + ddb(changes_to_dispose(), clear_changes_to_disposes())
+        process_dirty_topics();                           // all ddb
+        process_to_send_lists();                          // server + ddb(get_to_send, remove_to_send_this)
+        process_changes_release();                        // server + ddb(changes_to_release(), clear_changes_to_release())
+        result = pending_ack();                           // all server
```

Comment:
I think we need to explain why `result` comes from this function and what does it mean. Probably we can rename `result` to `pending_work` to make it clearer